ext/ddtrace.stub.php

```diff
@@ -0,0 +1,496 @@
+<?php
+
+/** @generate-class-entries */
+
+// phpcs:disable PSR1.Classes.ClassDeclaration.MultipleClasses
+
+namespace DDTrace {
+
+    class SpanData {
+        /**
+         * @var string The span name (Optional as of ddtrace v0.47.0; defaults to ‘ClassName.methodName’ if not set)
+         */
+        public string $name;
+
+        /**
+         * @var string The resource you are tracing (Optional as of ddtrace v0.47.0; defaults to SpanData::$name
+         * if not set)
+         */
+        public string $resource;
+
+        /**
+         * @var string The service you are tracing
+         */
+        public string $service;
+
+        /**
+         * @var string The type of request which can be set to: web, db, cache, or custom (Optional)
+         */
+        public string $type;
+
+        /**
+         * @var string[] $meta An array of key-value span metadata; keys and values must be strings (Optional)
+         */
+        public array $meta;
+
+        /**
+         * @var float[] $metrics An array of key-value span metrics; keys must be strings and values must be floats
+         * (Optional)
+         */
+        public array $metrics;
+
+        /**
+         * @var \Throwable $exception An exception generated during the execution of the original function, if any.
+         */
+        public \Throwable $exception;
+
+        /**
+         * @var string The unique identifier of the span
+         */
+        public string $id;
+
+        /**
+         * @var SpanData|null The parent span, or 'null' if there is none
+         */
+        public SpanData|null $parent;
+
+        /**
+         * @var SpanStack The span's stack trace
+         */
+        public SpanStack $stack;
+    }
+
+    class SpanStack {
+        /**
+         * @var SpanStack|null The parent stack, or 'null' if there is none
+         */
+        public SpanStack|null $parent;
+
+        /**
+         * @var SpanData The active span
+         */
+        public SpanData $active;
+    }
+
+    /**
+     * Close child spans of a parent span if a non-internal span is given,
+     * else if 'null' is given, close active, non-internal spans
+     * @param SpanData|null $span The parent span
+     * @return bool|int 'false' if spans couldn't be closed, else the number of span closed
+     */
+    function close_spans_until(?SpanData $span): bool|int {}
```

Comment:
`false` is actually also a valid return type
```suggestion
    function close_spans_until(?SpanData $span): false|int {}
```